bgpd/bgp_ecommunity.c

```diff
@@ -572,7 +678,24 @@ struct ecommunity *ecommunity_str2com(const char *str, int type,
 			if (ecom == NULL)
 				ecom = ecommunity_new();
 			eval.val[1] = type;
-			ecommunity_add_val(ecom, &eval, false, false);
+			ecommunity_add_val_internal(ecom, (void *)&eval, false, false,
+						    ecom->unit_size);
+			break;
+		case ecommunity_token_val6:
+			if (keyword_included) {
+				if (!keyword) {
+					if (ecom)
```

Comment:
seems this test is not needed, ecommunity_free() has such check itself

Reply:
remove it